src/model/guild/automod.rs

```diff
@@ -0,0 +1,859 @@
+#[serde(from = "u8", into = "u8")]
+#[non_exhaustive]
+pub enum TriggerType {
+    Keyword,
+    HarmfulLink,
+    Spam,
+    KeywordPresent,
+    Unknown(u8),
+}
+
+impl From<u8> for TriggerType {
+    fn from(value: u8) -> Self {
+        match value {
+            1 => Self::Keyword,
+            2 => Self::HarmfulLink,
+            3 => Self::Spam,
+            4 => Self::KeywordPresent,
+            _ => Self::Unknown(value),
+        }
+    }
+}
+
+impl From<TriggerType> for u8 {
+    fn from(value: TriggerType) -> Self {
+        match value {
+            TriggerType::Keyword => 1,
+            TriggerType::HarmfulLink => 2,
+            TriggerType::Spam => 3,
+            TriggerType::KeywordPresent => 4,
+            TriggerType::Unknown(unknown) => unknown,
+        }
+    }
+}
+
+/// Individual change for trigger metadata within an audit log entry.
+///
+/// Different fields are relevant based on the value of trigger_type.
+/// See [`Change::TriggerMetadata`].
+///
+/// [`Change::TriggerMetadata`]: crate::model::guild::audit_log::Change::TriggerMetadata
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct TriggerMetadata {
+    keyword_filter: Option<Vec<String>>,
+    presets: Option<Vec<KeywordPresentType>>,
+}
+
+/// Internally pre-defined wordsets which will be searched for in content.
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Deserialize, Serialize)]
+#[serde(from = "u8", into = "u8")]
+#[non_exhaustive]
+pub enum KeywordPresentType {
+    Profanity,
+    SexualContent,
+    Slurs,
+    Unknown(u8),
+}
+
+impl From<u8> for KeywordPresentType {
+    fn from(value: u8) -> Self {
+        match value {
+            1 => Self::Profanity,
+            2 => Self::SexualContent,
+            3 => Self::Slurs,
+            _ => Self::Unknown(value),
+        }
+    }
+}
+
+impl From<KeywordPresentType> for u8 {
+    fn from(value: KeywordPresentType) -> Self {
+        match value {
+            KeywordPresentType::Profanity => 1,
+            KeywordPresentType::SexualContent => 2,
+            KeywordPresentType::Slurs => 3,
+            KeywordPresentType::Unknown(unknown) => unknown,
+        }
+    }
+}
+
+/// An action which will execute whenever a rule is triggered.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Action {
+    /// Blocks the content of a message according to the rule.
+    BlockMessage,
+    /// Logs user content to a specified channel.
+    Alert(ChannelId),
+    /// Timeout user for a specified duration.
+    ///
+    /// Maximum of 2419200 seconds (4 weeks).
+    ///
+    /// A `Timeout` action can only be setup for [`Keyword`] rules.
+    /// [`Permissions::MODERATE_MEMBERS`] permission is required to use the `Timeout` action type.
+    ///
+    /// [`Keyword`]: TriggerType::Keyword
+    /// [`Permissions::MODERATE_MEMBERS`]: crate::model::Permissions::MODERATE_MEMBERS
+    Timeout(u64),
```

Comment:
`Timeout { duration_seconds: u64 }` perhaps? To make the unit clear (might be minutes or milliseconds as well)